models/_sector/nft/trades/ethereum/platforms/blur_v2_ethereum_base_trades.sql

```diff
@@ -0,0 +1,110 @@
+    , CAST(bitwise_right_shift(makerFeeRecipientRate, 160) AS double)/10000 AS fee
+    , from_hex('0x' || LOWER("RIGHT"(CAST(to_hex(CAST(makerFeeRecipientRate AS varbinary)) AS varchar), 40))) AS royalty_fee_address
+    FROM {{ source('blur_v2_ethereum','BlurPool_evt_Execution721MakerFeePacked') }}
+    {% if is_incremental() %}
+    WHERE evt_block_time >= date_trunc('day', now() - interval '7' day)
+    {% else %}
+    WHERE evt_block_time >= {{blur_v2_start_date}}
+    {% endif %}
+    
+    UNION ALL
+    
+    SELECT evt_tx_hash AS tx_hash
+    , CAST(ROUND((bitwise_right_shift(collectionPriceSide, 160) - (bitwise_right_shift(collectionPriceSide, 248) * CAST(power(2, 88) AS UINT256)))/POWER(10, 18), 8) AS double) AS price
+    , evt_block_time AS block_time
+    , evt_block_number AS block_number
+    , 'taker' AS fee_side
+    , evt_index
+    , contract_address AS project_contract_address
+    , CAST(bitwise_right_shift(collectionPriceSide, 248) AS BIGINT) AS order_type
+    , from_hex('0x' || LOWER("RIGHT"(CAST(to_hex(CAST(collectionPriceSide AS varbinary)) AS varchar), 40))) AS nft_contract_address
+    , orderHash AS order_hash
+    , bitwise_right_shift(tokenIdListingIndexTrader, 168) AS nft_token_id
+    , from_hex('0x' || LOWER("RIGHT"(CAST(to_hex(CAST(tokenIdListingIndexTrader AS varbinary)) AS varchar), 40))) AS trader
+    , CAST(bitwise_right_shift(takerFeeRecipientRate, 160) AS double)/10000 AS fee
+    , from_hex('0x' || LOWER("RIGHT"(CAST(to_hex(CAST(takerFeeRecipientRate AS varbinary)) AS varchar), 40))) AS royalty_fee_address
+    FROM {{ source('blur_v2_ethereum','BlurPool_evt_Execution721TakerFeePacked') }}
+    {% if is_incremental() %}
+    WHERE evt_block_time >= date_trunc('day', now() - interval '7' day)
+    {% else %}
+    WHERE evt_block_time >= {{blur_v2_start_date}}
+    {% endif %}
+    )
+
+SELECT CAST(date_trunc('day', bt.block_time) AS date) AS block_date
+, bt.block_time
+, bt.block_number
+, bt.tx_hash
+, bt.evt_index AS sub_tx_trade_id
+, CASE WHEN txs."from" = bt.trader THEN 'Sell' ELSE 'Buy' END AS trade_category
+, 'secondary' AS trade_type
+, CASE WHEN txs."from" = bt.trader THEN bt.trader ELSE txs."from" END AS buyer
+, CASE WHEN txs."from" = bt.trader THEN txs."from" ELSE bt.trader END AS seller
+, bt.nft_contract_address
+, CAST(bt.nft_token_id AS UINT256) AS nft_token_id
+, CAST(1 AS UINT256) AS nft_amount
+, CAST(bt.price*POWER(10, 18) AS UINT256) AS price_raw
+, CASE WHEN bt.order_type = 0 THEN 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2 ELSE 0x0000000000a39bb272e79075ade125fd351887ac END AS currency_contract
+, bt.project_contract_address
+, CAST(0 AS uint256) AS platform_fee_amount_raw
+, CAST(NULL AS varbinary) AS platform_fee_address
+, CAST(ROUND(bt.price * POWER(10, 18) * bt.fee) AS UINT256) AS royalty_fee_amount_raw
+, bt.royalty_fee_address
+FROM blur_v2_trades bt
+INNER JOIN {{ source('ethereum', 'transactions') }} txs ON txs.block_number=bt.block_number
```

Comment:
If possible, we should try and remove the join with transactions to make the query more optimized. 
Currently this is only used to determine:
- trade_category
- buyer
- seller

Is it possible to source these columns from the event data?